.github/workflows/run-simulators.yml

```diff
@@ -10,13 +10,50 @@ jobs:
       runs-on: ubuntu-latest
       concurrency:
         group: sim
+      outputs:
+        volume_id: ${{ steps.create_volume_step.outputs.volume_id }}
+      env:
+        INSTANCE_ID: ${{ secrets.AWS_EC2_INSTANCE_ID }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        AWS_DEFAULT_REGION: ${{ secrets.AWS_REGION }}
       steps:
+        - name: Create Volume from Latest Snapshot and Attach to Instance
+          id: create_volume_step
+          run: |
+            #  Retrieve the latest snapshot ID
+            LATEST_SNAPSHOT_ID=$(aws ec2 describe-snapshots --owner-ids self --query 'Snapshots | sort_by(@, &StartTime) | [-1].SnapshotId' --output text)
+            echo "Checking availability for snapshot: $LATEST_SNAPSHOT_ID"
+
+            # Wait until snapshot is in 'completed' status
+            while true; do
```

Comment:
Since you're using the nice `aws ec2 wait` command elsewhere, maybe we can simplify the code by replacing the manual polling loop with `aws ec2 wait snapshot-completed`?

Reply:
Updated the manual snapshot polling loop to use aws ec2 wait snapshot-completed.